src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Form.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
+{
+    public class Form
+    {
+        /// <summary>
+        /// Represents Form ID
+        /// </summary>
+        public int FormId;
+
+        /// <summary>
+        /// Title for the form 
+        /// </summary>
+        public string FormTitle;
+
+        /// <summary>
+        /// Represents list of inputs for the form
+        /// </summary>
+        public List<FormInput> FormInputs;
+
+        /// <summary>
+        /// Creates instance of Form class
+        /// </summary>
+        /// <param name="id">Unique ID for the Form</param>
+        /// <param name="title">Title for the Form</param>
+        public Form(int id, string title = "")
+        {
+            FormId = id;
+            FormInputs = new List<FormInput>();
+            CurrentInputIds = new HashSet<int>();
+            FormTitle = title;
+        }
+
+        /// <summary>
+        /// Hashset containing input IDs
+        /// </summary>
+        private HashSet<int> CurrentInputIds;
+
+        /// <summary>
+        /// Adds a form input to the form
+        /// </summary>
+        /// <param name="input">Input element to be added to the form</param>
+        public void AddFormInput(FormInput input)
+        {
+            // Returns true if we are able to add the input id to hashset
+            if(CurrentInputIds.Add(input.InputId))
+            {
+                input.CombinedId = input.InputId.ToString() + "." + FormId.ToString();
+                FormInputs.Add(input);             
+            }
+            else
+            {
+                throw new Exception($"Input ID {input.InputId} already exists for Form {FormId}. Please give a unique ID for the input");
+            }          
+        }
+
+        /// <summary>
+        /// Adds a list of form inputs to the form
+        /// </summary>
+        /// <param name="inputs">List of form inputs</param>
+        public void AddFormInputs(List<FormInput> inputs)
+        {
+            inputs.ForEach(input =>
+            {
+                AddFormInput(input);
+            });
+        }
+    }
+
+    public abstract class FormInput
+    {
+        /// <summary>
+        /// Represents Input ID
+        /// </summary>
+        public int InputId;
+
+        /// <summary>
+        /// Represents the Input Type
+        /// </summary>
+        public InputTypes InputType;
+
+        /// <summary>
+        /// Represents whether this is a required input
+        /// </summary>
+        public bool IsRequired;
+
+        /// <summary>
+        /// Represents the label of the input 
+        /// </summary>
+        public string Label;
+
+        /// <summary>
+        /// Combines the parent form id and input id
+        /// </summary>
+        public string CombinedId;
```

Comment:
should we call this as InternalId?

Reply:
As we discussed, this actually can be removed from the class and can be created on the UI side as its specific to the UI